pandas/io/excel/_odfreader.py

```diff
@@ -28,8 +28,28 @@ def __init__(
         storage_options: StorageOptions = None,
     ):
         import_optional_dependency("odf")
+        self._monkeypatch_odf_element_str()
         super().__init__(filepath_or_buffer, storage_options=storage_options)
 
+    def _monkeypatch_odf_element_str(self):
```

Comment:
woa, why don't you just change the impl to be correct? we *never* want to monkeypatch things in library code.

Reply:
I agree that we should not be monkeypatching the library code and ideally this should be fixed in odfpy directly.

If you look at the fix it essentially does the exact thing that the library does https://github.com/eea/odfpy/blob/master/odf/element.py#L240-L244 except add the case that PR #33233 addresses. This is relevant mostly because these Nodes can get specially if the nested child nodes also contain multiple spaces that #33233 addresses. Then it would become a recursive implementation. 

For a cell like this:
![image](https://user-images.githubusercontent.com/1978117/92432285-bb07d680-f167-11ea-9aeb-7d905b874f35.png)
there is no way to get all the spaces without a recursive implementation.

Reply:
right why can't you just change _get_string_cell_value to what you are doing here? (and put a nice comment / link to the source & issue)

Reply:
got it. done

not sure why the tests are failing. they seem unrelated.